spec/dry_spec.rb

```diff
@@ -0,0 +1,37 @@
+CONSTRUCTOR_TYPES = %i[permissive schema strict strict_with_defaults].freeze
+
+RSpec.describe Dry do
+  describe '.Struct' do
+    context 'constructor types' do
+      CONSTRUCTOR_TYPES.each do |constructor|
+        it "returns a struct with constructor type #{constructor}" do
+          struct_klass = Dry.Struct(constructor, name: 'strict.string')
+          expect(struct_klass.constructor_type).to eq constructor
+
+          struct = struct_klass.new(name: 'Test')
+          expect(struct.__attributes__).to eq(name: 'Test')
+        end
+      end
+    end
+
+    context 'initializer block' do
+      subject(:struct) do
+        Person = Dry.Struct do
+          constructor_type :strict
+
+          attribute :library, 'strict.string'
+          attribute :language, 'strict.string'
+        end
+      end
+
+      it 'sets the correct constructor type' do
+        expect(struct.constructor_type).to eq :strict
```

Comment:
It'd be better to check if it behaves like a strict struct instead of checking configuration.